language/vm/src/file_format.rs

```diff
@@ -1224,6 +1224,55 @@ pub enum Bytecode {
     ///
     /// ```..., u64_value(1), u64_value(2) -> ..., u64_value```
     Shr,
+    /// Create an empty vector.
+    ///
+    /// Stack transition:
+    ///
+    /// ```... -> ..., vec[]```
+    VecEmpty(SignatureIndex),
+    /// Return the length of the vector,
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., vector_reference -> ..., u64_value```
+    VecLen,
```

Comment:
I think that we should include a `SignatureIndex` on all of these operations, not just `Vec::empty`.
- This makes the operations more consistent with each other 
- A bytecode-to-native compiler will have a much easier time compiling vector operations if it doesn't have to repeat the type inference of the bytecode verifier. Attaching the type to the operations makes it simple to allocate the appropriate amount of space
- It reduces our reliance on the magic of type inference in the bytecode verifier

Reply:
Agreed